File: scripts/convert-encoding.py
import csv

input_path = r"c:\Users\노혜정\Downloads\product_recommendation_db_v2.csv"
output_path = r"c:\Users\노혜정\Downloads\giftfinder\data\products_utf8.csv"

# 디렉토리가 없으면 생성 (data 폴더)
import os
os.makedirs(os.path.dirname(output_path), exist_ok=True)

# CP949나 EUC-KR로 읽어보고 에러나면 다른 인코딩 시도
encodings = ['cp949', 'euc-kr', 'utf-8-sig', 'utf-8']
success = False

for enc in encodings:
    try:
        with open(input_path, 'r', encoding=enc) as f:
            lines = f.readlines()
        print(f"Successfully read with {enc}. Writing to {output_path} as UTF-8...")
        
        # 헤더 출력해서 제대로 읽었는지 확인
        print("Header sample:")
        print(lines[0].strip())
        print("First row sample:")
        print(lines[1].strip())
        
        with open(output_path, 'w', encoding='utf-8', newline='') as f_out:
            f_out.writelines(lines)
            
        success = True
        break
    except Exception as e:
        print(f"Failed with {enc}: {e}")

if not success:
    print("Failed to convert file encoding.")

File: scripts/convert-encoding.mjs
import { readFileSync, writeFileSync, mkdirSync } from "fs";
import { dirname } from "path";

const inputPath = "c:\\Users\\노혜정\\Downloads\\product_recommendation_db_v2.csv";
const outputPath = "c:\\Users\\노혜정\\Downloads\\giftfinder\\data\\products_utf8.csv";

try {
  // data 폴더 생성
  mkdirSync(dirname(outputPath), { recursive: true });

  // euc-kr 디코딩을 위해 파일 버퍼로 읽기
  const buffer = readFileSync(inputPath);
  const decoder = new TextDecoder("euc-kr");
  const text = decoder.decode(buffer);

  // 헤더 및 첫 줄 샘플 출력
  const lines = text.split("\n");
  console.log("=== Node.js 디코딩 성공 ===");
  console.log("Header sample:", lines[0]);
  console.log("First row sample:", lines[1]);

  // UTF-8로 저장
  writeFileSync(outputPath, text, "utf-8");
  console.log(`Saved UTF-8 file to: ${outputPath}`);
} catch (e) {
  console.error("Failed to decode or write file:", e);
}
